python/tvm/script/parser/tir/parser.py

```diff
@@ -199,16 +199,20 @@ def visit_with(self: Parser, node: doc.With) -> None:
 def visit_function_def(self: Parser, node: doc.FunctionDef) -> None:
     with self.var_table.with_frame():
         self.var_table.add("range", T.serial)
-        with T.prim_func():
-            T.func_name(node.name)
-            if node.returns is not None:
-                ret_type = self.eval_expr(node.returns)
-                if callable(ret_type):
-                    ret_type = PrimType(ret_type().dtype)
-                T.func_ret(ret_type)
-            with self.with_dispatch_token("tir"):
-                self.visit(node.args)
-                self.visit_body(node.body)
+        try:
```

Comment:
error handling logic should be moved to around this line: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L146.

Similarly, here: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L152; and here: https://github.com/apache/tvm/blob/ir-builder/python/tvm/script/parser/parser.py#L126